src/Container/ApplicationPipelineFactory.php

```diff
@@ -0,0 +1,22 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-expressive for the canonical source repository
+ * @copyright Copyright (c) 2018 Zend Technologies USA Inc. (https://www.zend.com)
+ * @license   https://github.com/zendframework/zend-expressive/blob/master/LICENSE.md New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace Zend\Expressive\Container;
+
+use Psr\Container\ContainerInterface;
+use Zend\Stratigility\MiddlewarePipe;
+use Zend\Stratigility\MiddlewarePipeInterface;
+
+class ApplicationPipelineFactory
+{
+    public function __invoke(ContainerInterface $container) : MiddlewarePipeInterface
```

Comment:
Hm... We can use here just `MiddlewarePipe` unless we'd like to have this class extensible and return different implementation of that interface?

Reply:
Bingo! This is to allow developers to extend the factory if desired; as long as what they return is a `MiddlewarePipeInterface`, the extension remains valid.